pkg/image/importer/importer_test.go

```diff
@@ -370,3 +371,44 @@ func TestPing(t *testing.T) {
 		t.Errorf("Expected ErrNotV2Registry, got %v", err)
 	}
 }
+
+func TestRetryFailure(t *testing.T) {
+	if !isDockerError(errcode.ErrorCodeUnauthorized, errcode.ErrorCodeUnauthorized) {
+		t.Fatal("not an error")
+	}
+
+	// do not retry on Manifests()
+	repo := &mockRepository{repoErr: fmt.Errorf("does not support v2 API")}
+	r := NewRetryRepository(repo, 1).(*retryRepository)
+	if m, err := r.Manifests(nil); m != nil || err != repo.repoErr || r.retries != 1 {
+		t.Fatalf("unexpected: %v %v %#v", m, err, r)
+	}
+
+	// do not retry on Manifests()
+	repo = &mockRepository{repoErr: errcode.ErrorCodeUnauthorized}
+	r = NewRetryRepository(repo, 4).(*retryRepository)
+	if m, err := r.Manifests(nil); m != nil || err != repo.repoErr || r.retries != 4 {
+		t.Fatalf("unexpected: %v %v %#v", m, err, r)
+	}
+
+	// do not retry
+	repo = &mockRepository{getByTagErr: fmt.Errorf("does not support v2 API")}
+	r = NewRetryRepository(repo, 4).(*retryRepository)
+	m, err := r.Manifests(nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if m, err := m.GetByTag("test"); m != nil || err != repo.getByTagErr || r.retries != 4 {
+		t.Fatalf("unexpected: %v %v %#v", m, err, r)
+	}
+
+	// retry four times
+	repo = &mockRepository{getByTagErr: errcode.ErrorCodeUnauthorized}
+	r = NewRetryRepository(repo, 4).(*retryRepository)
+	if m, err = r.Manifests(nil); err != nil {
+		t.Fatal(err)
+	}
+	if m, err := m.GetByTag("test"); m != nil || err != repo.getByTagErr || r.retries != 0 {
+		t.Fatalf("unexpected: %v %v %#v", m, err, r)
+	}
+}
```

Comment:
You're checking just `Manifests` method, it would be nice to add tests to others method as well. So we're sure it doesn't get broken. I can handle that post-merge, if you want to.